edi_oca/models/edi_exchange_record.py

```diff
@@ -149,6 +158,19 @@ def _constrain_edi_exchange_state(self):
                     _("Exchange state must respect direction!")
                 )
 
+    @api.constrains("secondary_record_ids")
+    def _check_secondary_record_models(self):
+        """
+        Check that all secondary records have the same model.
```

Comment:
Why? IMO we should support different kind of records.

Reply:
True, we also though about it in the beginning but were not sure how to handle then the action to open the related records. However, I can see you suggested 2 options to handle this so then we could allow different models :)